vault/resource_identity_mfa_duo_test.go

```diff
@@ -0,0 +1,86 @@
+package vault
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+
+	"github.com/hashicorp/terraform-provider-vault/internal/consts"
+	"github.com/hashicorp/terraform-provider-vault/internal/identity/mfa"
+	"github.com/hashicorp/terraform-provider-vault/testutil"
+)
+
+func TestIdentityMFADuo(t *testing.T) {
+	t.Parallel()
+
+	resourceName := mfa.ResourceNameDuo + ".test"
+
+	checksCommon := []resource.TestCheckFunc{
+		resource.TestCheckResourceAttrSet(resourceName, consts.FieldUUID),
+		resource.TestCheckResourceAttrSet(resourceName, consts.FieldMethodID),
+		resource.TestCheckResourceAttr(resourceName, consts.FieldType, mfa.MethodTypeDuo),
+		resource.TestCheckResourceAttr(resourceName, consts.FieldNamespaceID, "root"),
+	}
+
+	res, err := mfa.GetDuoSchemaResource()
```

Comment:
It's not fully clear to me why we are doing this paired with the switch case. It looks like we're getting the full Duo resource (along with all context functions), mostly to add empty string checks for `name` and `mount_accessor`, and that too _if_ the Duo schema has the Name and Mount Acc. field on it. But since we're using `getMethodSchemaResource`, don't we always know that the resource schema will have those fields on it?

It seems like we can achieve the same behavior by simply appending these checks without getting the Duo schema resource and checking a known case with a switch case block:
```
resource.TestCheckResourceAttr(resourceName, consts.FieldName, "")
resource.TestCheckResourceAttr(resourceName, consts.FieldMountAccessor, "")
```

Reply:
I also see that this is behavior specific to Okta and Duo and not the other MFA resources in this PR. Maybe we're using this to add a guard for when the schema for Duo and Okta changes? 🤔 Let me know if I'm missing something in my understanding

Reply:
Fixed in 75bc3995bb942b958c52cd71e3b23c1eed9a9d8e